pennylane/gradients/parameter_shift_hessian.py

```diff
@@ -135,47 +198,24 @@ def expval_hessian_param_shift(
 
     hessian_tapes = []
     hessian_coeffs = []
-    for i in range(h_dim):
-        # The diagonal recipe is None if the parameter is not trainable or not in argnum
-        if diag_recipes[i] is None:
-            hessian_coeffs.extend([None] * (h_dim - i))
+    for i, j in it.combinations_with_replacement(range(h_dim), r=2):
+        if i == j:
+            add_unshifted, c_unshifted[(i, i)] = _generate_diag_tapes(
+                tape, i, diag_recipes, add_unshifted, hessian_tapes, hessian_coeffs
+            )
             continue
```

Comment:
Does this mean the diagonal elements are not masked out (as done below with `if not argnum[i, j]`) ?

Reply:
Yes they are but within `_generate_diag_tapes` :facepalm: Will definitely move that out...